app/src/main/java/com/example/cookbook/data/repositories_impl/OwnRecipeRepositoryImpl.kt

```diff
@@ -17,30 +16,38 @@ class OwnRecipeRepositoryImpl @Inject constructor(
     private val database: OwnRecipesDatabaseSource
 ) : OwnRecipeRepository {
 
-    override suspend fun getRecipeList(): List<RecipeData> {
-        return withContext(Dispatchers.IO) {
-            database.getAllRecipes().map { entityToDataMapper(it) }
+    override fun getRecipeList(): Single<List<RecipeData>> {
+        return database.getAllRecipes().map { entityList ->
+            entityList.map { entity ->
+                entityToDataMapper(entity)
+            }
         }
     }
 
-    override suspend fun getRecipeListSync(): LiveData<List<RecipeData>> {
-        return Transformations.map(database.getAllRecipesSync()) {
+    override fun getRecipeListSync(): Observable<List<RecipeData>> {
+        return database.getAllRecipesSync().map {
             it.map { entity ->
                 entityToDataMapper(entity)
             }
         }
     }
 
-    override suspend fun addNewRecipe(recipe: RecipeData) {
-        return withContext(Dispatchers.IO) {
-            val entity = dataToEntityMapper(recipe)
-            database.insertAllRecipes(entity)
-        }
+    override fun addNewRecipe(recipe: RecipeData): Completable {
+        return Single.just(recipe)
```

Comment:
I think we can get rid of Single here
You can use `Completable.fromAction`

Reply:
ok